### PR TITLE
[1512] Onboarding Popup

### PR DESCRIPTION
**Pull request by ihdk**
**Changes proposed in this Pull Request**
Implemented onboarding screen for first app set up.

Checking if onboarding should be displayed will need improvements once will be fixed issue with multiple loading of app. 
I left comments in the code to describe current implementation.

**Testing instructions**
- onboarding screen is displayed when user do not have inserted api key.
- remove api key from wp options table for test, meta key `urlslab-api-key`

**Checklist**
- [x] My code follows the style guidelines of this project
- [x] My commits follow the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

Close QualityUnit/web-issues#1512

### CODE REVIEW

File: admin/src/App.jsx [hunk 1]
```diff
@@ -44,9 +49,11 @@ export default function App() {
 
 				const isApiObject = generalData?.filter( ( dataset ) => dataset.id === 'api' )[ 0 ];
 				const hasApiKey = isApiObject?.options[ 'urlslab-api-key' ].value;
-
 				if ( ! hasApiKey ) {
+					setActiveOnboarding( true );
 					update( 'apiKeySet', ( ) => false );
+				} else {
```
Comment (jremen) [inline admin/src/App.jsx:55]: Would avoid else statement and rather use it non wrapped with return false in if condition

File: admin/src/assets/styles/layouts/_Onboarding.scss [hunk 2]
```diff
@@ -0,0 +1,272 @@
+@use "../common/mixins" as *;
+@use "../common/variables" as *;
+@forward "./Settings";
+@forward "../elements/Inputs";
+
+.urlslab-onboarding {
+
+	&-header {
+		width: 100%;
+		padding: 16px 25px;
```
Comment (jremen) [inline admin/src/assets/styles/layouts/_Onboarding.scss:10]: use relative units, em in this case (rem for font size)

File: admin/src/assets/styles/layouts/_Onboarding.scss [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+		margin-bottom: 40px;
```
Comment (jremen) [inline admin/src/assets/styles/layouts/_Onboarding.scss:11]: em units

File: admin/src/assets/styles/layouts/_Onboarding.scss [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+		background-color: $white;
+
+		&-logo {
+			left: 25px;
```
Comment (jremen) [inline admin/src/assets/styles/layouts/_Onboarding.scss:15]: em units

File: admin/src/assets/styles/layouts/_Onboarding.scss [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+
+				&:not(:last-child) {
+					margin-right: 26px;
+				}
+
+				&-icon {
+
+					@include square(40px);
+
+					font-size: 16px;
```
Comment (jremen) [inline admin/src/assets/styles/layouts/_Onboarding.scss:41]: 1rem or use fs-normal class in the tag

File: admin/src/assets/styles/layouts/_Onboarding.scss [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+					color: #b0b4bb;
+					border: 1px solid currentColor;
+					margin-right: 9px;
+					border-radius: 100%;
+					transition: all $transitionTime;
+				}
+
+				&-label {
+					font-size: 13px;
```
Comment (jremen) [inline admin/src/assets/styles/layouts/_Onboarding.scss:50]: rem units or use fs-xm class

File: admin/src/assets/styles/layouts/_Onboarding.scss [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+			}
+		}
+	}
+
+
+	&-content {
+		margin: 0 20px 40px;
+
+		&-wrapper {
+			width: 100%;
```
Comment (jremen) [inline admin/src/assets/styles/layouts/_Onboarding.scss:93]: I don't think width: 100% is needed if it's div

File: admin/src/assets/styles/layouts/_Onboarding.scss [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+			margin: 0 auto;
+
+			&.small-wrapper {
+				max-width: 500px;
```
Comment (jremen) [inline admin/src/assets/styles/layouts/_Onboarding.scss:97]: em units

File: admin/src/assets/styles/layouts/_Onboarding.scss [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+
+				path {
+					fill: currentColor
+				}
+			}
+		}
+
+		.urlslab-half-columns {
+			display: flex;
+			flex-direction: row;
```
Comment (jremen) [inline admin/src/assets/styles/layouts/_Onboarding.scss:238]: not needed if not specified in media query, default is row

File: admin/src/assets/styles/layouts/_Onboarding.scss [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+					width: 100%;
+					max-width: 100%;
+				}
+			}
+
+			@media (max-width: $breakpoint-mobile) {
+				flex-direction: column;
+
+				&-col {
+					width: 100%;
```
Comment (jremen) [inline admin/src/assets/styles/layouts/_Onboarding.scss:265]: not needed

**Review by jremen — CHANGES_REQUESTED** [review]
Seems ok. Just change units in CSS to relative ones and remove unneccessary width statements

File: admin/src/assets/styles/layouts/_Onboarding.scss (hunk 10 above)
Comment (ihdk) [reply admin/src/assets/styles/layouts/_Onboarding.scss:265]: this one should be there, otherwise it's 50% width from outside @media

**Comment by ihdk** [comment]
@jremen comments fixed

**Review by jremen — APPROVED** [review]

**Pull request merged**